Grant launch permission for multiple recent AMIs

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the post-Packer code to grant launch permission for several of the most recent AMIs.  Previously launch permission was granted only for _the_ most recent AMI.

This entails:
* Adding an input variable to specify X, the number of most-recent AMIs.
* Modifying `main.tf` to grant launch permission for the X most recent AMIs.
* Modifying the output to return the resulting array of [cisagov/ami-launch-permission-tf-module](https://github.com/cisagov/ami-launch-permission-tf-module) objects.

## 💭 Motivation and context ##

This is a necessary change because assessments can go on for several months, and [cisagov/guacscanner](https://github.com/cisagov/guacscanner) needs to be able to view AMI information in order to create Guacamole connections.  If Guacamole is re-deployed mid-assessment, then [cisagov/guacscanner](https://github.com/cisagov/guacscanner) may need to re-create connections for old AMIs.

It is important to grant full launch permissions for these historical AMIs, since some assessment types will want to launch additional instances using the same AMI from which their original instances were launched.

Resolves cisagov/cool-system-internal#47.

## 🧪 Testing ##

I applied the new code to the staging Terraform workspace and verified that it created the expected resources.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

jsf9k committed:
Grant launch permission for the X most-recent AMIs

This entails:
* Adding an input variable specifying X, the number of most-recent
AMIs
* Modifying main.tf to grant launch permission for the X most recent
AMIs
* Modifying the output to return the resulting array of
cisagov/ami-launch-permission-tf-module objects.

This is a necessary change because assessments can go on for several
months, and cisagov/guacscanner needs to be able to view AMI
information in order to create Guacamole connections.  If Guacamole is
re-deployed mid-assessment, then cisagov/guacscanner may need to
re-create connections for old AMIs.

It is important to grant full launch permissions for these historical
AMIs, since some assessment types will want to launch additional
instances using the same AMI from which their original instances were
launched.

See also cisagov/cool-system-internal#47.

dav3r commented on terraform-post-packer/main.tf:
```diff
@@ -43,12 +43,16 @@ data "aws_ami" "example" {
 module "ami_launch_permission" {
+  # Really we only want the var.recent_ami_count most recent AMIs, but
+  # we have to cover the case where there are fewer than that many
+  # AMIs in existence.  Hence the max()/length() tomfoolery.
```
```suggestion
  # AMIs in existence.  Hence the min()/length() tomfoolery.
```

jsf9k commented:
I am requesting to merge this PR outside of our usual Kraken run since it is necessary for the current UAT exercise.

jsf9k committed:
Correct comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r commented on terraform-post-packer/variables.tf:
```diff
@@ -9,3 +9,9 @@ variable "extraorg_account_ids" {
+
+variable "recent_ami_count" {
+  type        = number
+  description = "The number of most-recent AMIs for which to grant launch permission (e.g. \"3\").  If this variable is set to three, for example, then accounts will be granted permission to launch the three most recent AMIs."
```
Do we want to put any language in here about the case where there are fewer existing AMIs than `recent_ami_count`?

jsf9k replied:
Please accept commit 77378c7 as tribute.

dav3r reviewed (approved):
This looks great to me! 👍

mcdonnnj commented on terraform-post-packer/main.tf:
```diff
@@ -24,8 +24,8 @@ locals {
 
-# The most-recent AMI created by cisagov/skeleton-packer
-data "aws_ami" "example" {
+# The IDs of all cisagov/skeleton-packer AMIs
```
The ID is an attribute of an AMI resource.

```suggestion
# The AMIs created by cisagov/skeleton-packer
```

jsf9k replied:
[`aws_ami`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami) is a different data resource than [`aws_ami_ids`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami_ids).  The latter returns only the IDs, not the complete AMI object.

mcdonnnj replied:
Ack I missed the resource type change, my apologies.

mcdonnnj reviewed (approved):
Nice addition. I had one suggestion related to comment language for your consideration.

jsf9k committed:
Update variable description

This is to address the case where there are fewer AMIs available than
requested.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
